### PR TITLE
Support specified columns for full-text index

### PR DESCRIPTION
**Pull request by bright-starry-sky**

### CODE REVIEW

**Comment by bright-starry-sky** [comment]
Ready to review.

File: src/common/interface/meta.thrift [hunk 1]
```diff
@@ -1033,6 +1033,32 @@ struct ListFTClientsResp {
     3: list<FTClient>      clients,
 }
 
+struct FTIndex {
```
Comment (critical27) [inline src/common/interface/meta.thrift:1036]: How to make sure that there won't be more than 1 FTIndex of a given space and schema? And why we make this assumption?

> getFTIndexBySpaceSchemaFromCache

File: src/common/interface/meta.thrift [hunk 2]
```diff
@@ -1033,6 +1033,32 @@ struct ListFTClientsResp {
...
+    1: common.GraphSpaceID  space_id,
+    2: SchemaID             depend_schema,
+    3: list<binary>         fields,
```
Comment (critical27) [inline src/common/interface/meta.thrift:1039]: A question here, related to previous comment, what does the `fields` mean here? Is it several fields of a new FTIndex, or all fields that need to build FTIndex of a given tag/edge.

File: src/common/interface/meta.thrift (hunk 1 above)
Comment (bright-starry-sky) [reply src/common/interface/meta.thrift:1036]: > 1 FTIndex of a given space and schema

Only one index can be created by space and schema.

Comment (critical27) [reply src/common/interface/meta.thrift:1036]: 👌 , see comment in storage

File: src/common/interface/meta.thrift (hunk 2 above)
Comment (bright-starry-sky) [reply src/common/interface/meta.thrift:1039]: > A question here, related to previous comment, what does the `fields` mean here? Is it several fields of a new FTIndex, or all fields that need to build FTIndex of a given tag/edge.
> 
> According to storage code, I think it is the latter one?

Aha, this `fields` means A collection of full-text index, allowing one or more fields, for example :
CREATE TAG player(c_int int, c_string string, c_fixed_string_256 fixed_string(256));
CREATE FULLTEXT INDEX nebula_ft_idx ON TAG player(c_string, c_fixed_string_256); 

the fields are  : (c_string , c_fixed_string_256)

Comment (yixinglu) [reply src/common/interface/meta.thrift:1039]: prefer to create syntax:

```
CREATE FULLTEXT TAG INDEX player_ft_index ON player(c_string, c_fixed_string_256);
```

Comment (bright-starry-sky) [reply src/common/interface/meta.thrift:1039]: > prefer to create syntax:
> 
> ```
> CREATE FULLTEXT TAG INDEX player_ft_index ON player(c_string, c_fixed_string_256);
> ```

Good advice. let me to improve it in graph layer.

**Review by critical27 — APPROVED** [review]
LGTM

**Review by panda-sheep — APPROVED** [review]
Good job!

**Pull request merged**